application/models/Question.php

```diff
@@ -1573,6 +1576,9 @@ public function getQuestionTheme()
         if (empty($questionThemeName) || $questionThemeName == 'core') {
             $questionThemeName = QuestionTheme::model()->getBaseThemeNameForQuestionType($this->type);
         }
-        return QuestionTheme::model()->findByAttributes(['question_type' => $this->type, 'name' => $questionThemeName]);
+        if (empty($this->relatedQuestionTheme) || $this->relatedQuestionTheme->name != $questionThemeName) {
+            $this->relatedQuestionTheme = QuestionTheme::model()->findByAttributes(['question_type' => $this->type, 'name' => $questionThemeName]);
```

Comment:
This should use the new relation instead?

Reply:
We are thinking of not using the relation and discouraging its usage here.
Use this method getQuestionTheme() instead.

More explained here:
https://github.com/LimeSurvey/LimeSurvey/blob/5d5c2a732d0765bd61adeedc60e483a7d7778e18/application/models/Question.php#L130

Reply:
Shouldn't $question->questionTheme rather return a default theme if $question is not yet saved? Is that possible?

Reply:
hmm we could make something about it.
We could have the method to return the theme Associated to the question type. if no question type is set we could use the default question type or the default theme

Shall we?

Reply:
Or, it's not enough for the relation to set question_type and question_theme_name before using the relation?

```php
$question = new Question(); // Empty object
$question->question_type = 'L';
$question->question_theme_name = 'blaha';
$theme = $question->question_theme;
```

Reply:
When adding a new record the relationship returns null
![image](https://user-images.githubusercontent.com/4942495/123453913-5d3d6180-d5b6-11eb-81e1-3aea80b60daa.png)

Reply:
Interesting.

Reply:
So it _would_ be possible to use the relation if we use the `getRelated` method and set `$refresh` to `true`?

Reply:
Instead of `$question->question_theme`, it would have to be `$question->getRelated('question_theme', true);`

Reply:
No reason to add a new method if one already exist.

Reply:
Great!

Reply:
Alternatively, getQuestionTheme() just calls $this->getRelated("question_theme", true);

Reply:
Eventually, we could, but not sure it is worth it right now.

Reply:
Why not worth it? It's one line. :)

Reply:
We are reviewing and analyzing. Will be back soon. Thanks